### PR TITLE
Add eslint-plugin-perfectionist to the API package

### PR DESCRIPTION
**Pull request by richardcane**
Implements https://perfectionist.dev/configs/recommended-natural as a way to sort and order various bits of our codebase, including things like function parameters, object properties and imports.

The natural configuration for perfectionist intends to improve the intuitiveness of the code to reduce cognitive load and strain.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -76,6 +72,15 @@ module.exports = {
         "@typescript-eslint/no-explicit-any": "off",
         "jest/no-standalone-expect": "off"
       }
+    },
+    {
+      extends: ["plugin:perfectionist/recommended-natural-legacy"],
+      files: "packages/+(api)/**/*",
```
Comment (chubberlisk) [inline .eslintrc.js:78]: Just an understanding question: what does this the + and () mean in `+(api)` mean? Is it "only in the api directory" and then when we do the other packages it'll be like `+(api|core)` or something?

**Review by chubberlisk — APPROVED** [review]
Thank you for breaking down the original PR, this was much quicker and easier to review! ✅

File: .eslintrc.js (hunk 1 above)
Comment (richardcane) [reply .eslintrc.js:78]: https://www.digitalocean.com/community/tools/glob

Comment (richardcane) [reply .eslintrc.js:78]: + is the one or more matcher

Comment (richardcane) [reply .eslintrc.js:78]: For this PR I could have just used `packages/api/**/*` but I was messing with that to make sure the glob pattern was picking up the right packages so thought I'd leave it in as I'm going to be adding to it shortly anyway :D

Comment (chubberlisk) [reply .eslintrc.js:78]: Okayyy, that makes sense - thanks (and also for the glob tool, didn't know that was a thing)!

**Pull request merged**